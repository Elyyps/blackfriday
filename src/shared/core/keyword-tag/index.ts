export * from './keyword-tag.component'