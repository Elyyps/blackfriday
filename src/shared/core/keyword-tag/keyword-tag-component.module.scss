.keyword-tag {
  padding: 8px;
  border: 1px solid #e0e0e0;
  font-family: Assistant;
  font-size: 10px;
  font-weight: bold;
  font-style: normal;
  font-stretch: normal;
  line-height: 1.29;
  letter-spacing: normal;
  color: #777;
}
