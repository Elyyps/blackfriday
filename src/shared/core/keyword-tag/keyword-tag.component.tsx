import * as React from "react";
import styles from "./keyword-tag-component.module.scss";
import { LinkComponent } from "../link";

export interface IKeywordTagComponentProps {
  children: any;
  style?: object;
}

const KeywordTagComponent = (props: IKeywordTagComponentProps) => (
  <div className={styles["keyword-tag"]}>
    <LinkComponent to={"/stores-overview"}>{props.children}</LinkComponent>
  </div>
);

export { KeywordTagComponent };
