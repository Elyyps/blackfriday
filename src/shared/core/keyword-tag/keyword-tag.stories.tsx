
import * as React from "react";

import { storiesOf } from "@storybook/react";
import { withA11y } from "@storybook/addon-a11y";
import { withProvider } from "@app/util";
import { KeywordTagComponent } from "./keyword-tag.component";

storiesOf("KeywordTag", module)
  .addDecorator(withA11y)
  .addDecorator(withProvider)
  .add("Basic implementation", () => (
    <KeywordTagComponent />
  ));

  