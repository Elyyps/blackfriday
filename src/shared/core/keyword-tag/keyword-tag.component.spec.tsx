
import React from "react";
import { shallow } from "enzyme";
import { KeywordTagComponent } from "./keyword-tag.component";

describe("[KeywordTag]", () => {
  it("should render component without crashing", () => {
    const renderedComponent = shallow(
      <KeywordTagComponent />
    );
    expect(renderedComponent).toMatchSnapshot();
  });
});

  