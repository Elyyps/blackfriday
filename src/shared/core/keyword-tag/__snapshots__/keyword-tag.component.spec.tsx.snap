// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[KeywordTag] should render component without crashing 1`] = `
<div
  className="keyword-tag"
>
  <LinkComponent
    to="/stores-overview"
  >
     Tag Content 
  </LinkComponent>
</div>
`;
